include/diy/master.hpp

```diff
@@ -1039,6 +1064,19 @@ flush()
   unsigned wait = 1;
 #endif
 
+  // prepare for next round
+  ++exchange_round_;
+  if (!incoming_next_.empty() && incoming_next_.begin()->first == exchange_round_)
```

Comment:
I think this can be simplified. `incoming_next_[exchange_round]` should give the correct instance, possibly initialized to zero.